src/main/java/games/strategy/internal/persistence/serializable/VersionProxy.java

```diff
@@ -0,0 +1,56 @@
+package games.strategy.internal.persistence.serializable;
```

Comment:
Sure, I'm open to any suggestion that improves the package structure of the proxies for the reasons you gave.  However, if by "close," you mean in the same file or package, I would tend to disagree with that.  Again, thinking about the code broken up into modules, I don't think the domain module should have any knowledge of or dependency on the persistence module.  Rather, the persistence module should depend on the domain module.  The simplified diagram below kind of represents my thinking here (where the UI module not only contains the UI code, but also acts as an orchestration layer, which is why it depends on the persistence module):

![modules](https://user-images.githubusercontent.com/4826349/27706153-7145db08-5cde-11e7-95b0-a0eb13319318.png)

Here are some other options I considered for the proxy package structure:

#### Near-fully-qualified subpackages

In this option, we would attempt to mirror the domain class package structure underneath some persistence-layer-specific root package.  As long as the dev knows the root package for a particular persistence layer, they would know how to drill down to find the proxy for the domain class they're looking for.  In the example below, I use the near-fully-qualified package name of the domain class (excluding `games.strategy` because it seems redundant):

```
src/main/java/games/strategy/
  internal/
    persistence/
      serializable/
        proxies/
          engine/
            data/
              GameDataPersistenceDelegate.java
              GameDataProxy.java
          util/
            memento/
              PropertyBagMementoPersistenceDelegate.java
              PropertyBagMementoProxy.java
            VersionPersistenceDelegate.java
            VersionProxy.java
```

#### Different source sets

In this option, we would create a new source set for each persistence layer so the package structure would parallel the domain class package structure, similar to what we do for tests.  It would be easy for a dev to locate the corresponding proxies because they are already familiar with this pattern when they have to locate a test.  For example:

```
src/main/java/games/strategy/
  . . .
src/persistence-serializable/java/games/strategy/
  engine/
    data/
      GameDataPersistenceDelegate.java
      GameDataProxy.java
    util/
      memento/
        PropertyBagMementoPersistenceDelegate.java
        PropertyBagMementoProxy.java
      VersionPersistenceDelegate.java
      VersionProxy.java
```

One drawback of this solution is that the proxies end up being in the same package as the domain classes at runtime.  Even though this violates the preference I expressed above, I can live with it because it at least keeps the concerns separate in the file structure that the dev sees, which should still encourage a separation between the domain and persistence modules.  Until TripleA adopts a module framework like OSGi or Jigsaw, "separation of modules" is effectively virtual and simply enforced by programmer discipline.

Another drawback is if we ever have to support a second persistence layer concurrently (e.g. while we transition from `Serializable` to a JSON-based format), we may run into name collisions.  For example, if in an imaginary JSON-based persistence layer, we also wanted to name the proxy classes with a `*Proxy` suffix, they would collide with the existing `Serializable` proxies.  I think it's a bit of a bad smell if we then were forced to prefix the proxy names to distinguish them (e.g. `SerializableGameDataProxy` vs. `JsonGameDataProxy`).